packages/core/src/components/RadioButton/RadioButton.tsx

```diff
@@ -36,17 +37,15 @@ const RadioButton: React.FC<RadioButtonProps> = ({
 }) => {
   const { value: contextValue, onValueChange } = useRadioButtonGroupContext();
 
-  const handlePress = () => {
-    onPress && onPress();
+  const realValue = getValueForRadioButton(value);
+  const realContextValue = getValueForRadioButton(contextValue);
+  const isSelected = selected ?? realContextValue === realValue;
```

Comment:
this is a different operation than what used to be here - is that on purpose?

```
const isSelected = selected || (contextValue != null && contextValue === value);

// const isSelected = selected || (contextValue != null && contextValue === value);

```

```
const isSelected = selected ?? realContextValue === realValue;

// const isSelected = selected !== null && selected !== void 0 ? selected : realContextValue === realValue;
```

Reply:
@peterpme this is the original code...

`  const handlePress = () => {
    onPress && onPress();

    if (onValueChange && value) {
      onValueChange(value);
    }
  };

  const isSelected =
    selected || (contextValue != null && contextValue === value);`
    
... firstly, onPress will be called if it is passed in / exists now .. that is what this does: `onPress?.(realValue);`

Before you were not passing the value into onPress ... that looked like a mistake to me.  It looked like if you provided an  additional onPress, you would want to pass the value as well, but I can remove that if it is wrong.

For the `onValueChange && value` conditional, this is already assumed because we are initializing `onValueChange` so it IS there and we would have thrown if realValue was `null` or `undefined`.

Finally, isSelected is a simple comparison of non-null/non-undefined string values which we know exist because of `getValueForRadioButton`.  So we can compare them directly.  Consider also we want you to be able to compare empty strings as values in case for some reason your RadioButton is a reset.  So we do not need to verify a truthy string either.